File: FireBaseProject/app/src/main/java/com/example/firebaseproject/MainActivity.java
package com.example.firebaseproject;

import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;


public class MainActivity extends AppCompatActivity {

    Button markPresent, markAbsent;
    EditText name;
    private DatabaseReference databaseReference;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        markPresent = findViewById(R.id.markPresent);
        markAbsent = findViewById(R.id.markAbsent);
        name = findViewById(R.id.name);
        databaseReference = FirebaseDatabase.getInstance().getReference();

        markPresent.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String getName = name.getText().toString();
                databaseReference.child(getName).setValue("Present");
                name.setText("");
            }
        });

        markAbsent.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String getName = name.getText().toString();
                databaseReference.child(getName).setValue("Absent");
                name.setText("");
            }
        });

    }
}

File: kotlinDemo/app/src/main/java/com/example/kotlindemo/MainActivity.kt
package com.example.kotlindemo

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.widget.LinearLayout
import android.widget.Toast

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)



        val counter = findViewById(R.id.counter) as LinearLayout
        val bmi = findViewById(R.id.bmi) as LinearLayout
        val calculator = findViewById(R.id.calculator) as LinearLayout


        counter.setOnClickListener {
            intent = Intent(applicationContext, ActCounter::class.java)
            startActivity(intent)
        }

        bmi.setOnClickListener {
            intent = Intent(applicationContext, ActBmi::class.java)
            startActivity(intent)
        }

        calculator.setOnClickListener {
            intent = Intent(applicationContext, ActCalculator::class.java)
            startActivity(intent)
        }
    }

}